### PR TITLE
vt-uploader: Add version 0.1.9

### LINKED ISSUE #9349: [Request]  VirusTotalUploader (unofficial)
### Criteria

For a package to be acceptable in the Extras bucket, it should be:

- [X] a well-known and widely used tool (e.g. if it's a GitHub project, it should have at least 100 stars and 50 forks)
- [X] the latest stable version of the program

### Information

**Name:** VirusTotalUploader (unofficial)

**Description:**  C# Open-Source Winforms application for uploading files to VirusTotal

**Homepage:** https://github.com/SamuelTulach/VirusTotalUploader

**Download link(s):** https://github.com/SamuelTulach/VirusTotalUploader/releases

**Some indication of popularity/repute:** more than 700 stars and 105 forks. Suggested on the official VirusTotal website: https://support.virustotal.com/hc/en-us/articles/115002179065-Desktop-apps

P.S.: I am not sure the portable version adds a context menu option to scan files like installing does

### PR DESCRIPTION
**Pull request by Zliced13**
Closes #9349

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Zliced13** [comment]
@HUMORCE could you please test the manifest in my personal bucket? I need to know why the context menu I created for this manifest is not functioning properly?
```
scoop bucket add MyScoop 'https://github.com/Zliced13/MyScoop'; scoop install MyScoop/vt-uploader
```

**Comment by HUMORCE** [comment]
> why the context menu I created for this manifest is not functioning properly?

check the generated registry file finally.

btw:

1.why not use the package of portable one.
2.you should let the user choose whether to install the context menu or not, rather than forcing users to elevate privileges to do it, context menu is not a mandatory part.

**Comment by Zliced13** [comment]
If I get rid of the elevated privileges and forced context menu (make it optional). Would you help me figure out my context menu problem?

**Comment by HUMORCE** [comment]
> Then would you help me figure out my context menu problem?

I have already told you for this...

**Comment by HUMORCE** [comment]
> If I get rid of the elevated privileges and forced context menu (make it optional).

context menu without require elevated privileges is OK.

File: bucket/vt-uploader.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        "   New-Item \"$dir\\vtu_settings.json\" -ItemType 'File' -Value '{\"ApiKey\":\"\",\"Language\":\"\",\"DirectUpload\":false}' | Out-Null",
+        "}"
+    ],
+    "post_install": [
+        "Copy-Item \"$bucketsdir\\extras\\scripts\\vt-uploader\\install-context.reg\" \"$dir\\install-context.reg\"",
+        "$vtPath = $dir.Replace('\\', '\\\\')",
+        "$regContent = (Get-Content \"$dir\\install-context.reg\").Replace('$dir', \"$vtPath\")",
+        "Set-Content \"$dir\\install-context.reg\" -Value \"$regContent\" -Encoding 'Ascii'"
+    ],
+    "bin": "uploader.exe",
```
Comment (HUMORCE) [inline bucket/vt-uploader.json:27]: `uploader` is a common name, use `vt-uploader` or `virustotal-uploader`

**Comment by HUMORCE** [comment]
```
Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\*\shell\Upload to VirusTotal\command]
@="\"$dir\\uploader.exe\" \"%1\""

```


```
Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\*\shell\Upload to VirusTotal\command]
@="\"$diruploader.exe\" \"%1\""


```

File: bucket/vt-uploader.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        "if (!(Test-Path \"$persist_dir\\vtu_settings.json\")) {",
+        "   New-Item \"$dir\\vtu_settings.json\" -ItemType 'File' -Value '{\"ApiKey\":\"\",\"Language\":\"\",\"DirectUpload\":false}' | Out-Null",
+        "}"
+    ],
+    "post_install": [
+        "Copy-Item \"$bucketsdir\\extras\\scripts\\vt-uploader\\install-context.reg\" \"$dir\\install-context.reg\"",
+        "$vtPath = $dir.Replace('\\', '\\\\')",
+        "$regContent = (Get-Content \"$dir\\install-context.reg\").Replace('$dir', \"$vtPath\")",
+        "Set-Content \"$dir\\install-context.reg\" -Value \"$regContent\" -Encoding 'Ascii'"
+    ],
```
Comment (HUMORCE) [inline bucket/vt-uploader.json:26]:
```suggestion
    "post_install": [
        "$vtPath = $dir.Replace('\\', '\\\\')",
        "$regContent = (Get-Content \"$bucketsdir\\nuke\\scripts\\vt-uploader\\install-context.reg\").Replace('$dir', \"$vtPath\")",
        "Set-Content \"$dir\\install-context.reg\" -Value $regContent -Encoding 'Ascii'",
    ],
```

**Comment by Zliced13** [comment]
Thank you, @HUMORCE, the context menu now works perfectly.

**Comment by Zliced13** [comment]
I used the setup package, because that one contains language files that could be helpful for non-English speakers. The portable version does not.

File: bucket/vt-uploader.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        "Rename-Item \"$dir\\uploader.exe\" 'setup.exe'; Expand-InnoArchive \"$dir\\setup.exe\" \"$dir\" -Removal",
+        "if (!(Test-Path \"$persist_dir\\vtu_settings.json\")) {",
+        "   New-Item \"$dir\\vtu_settings.json\" -ItemType 'File' -Value '{\"ApiKey\":\"\",\"Language\":\"\",\"DirectUpload\":false}' | Out-Null",
+        "}"
+    ],
+    "post_install": [
+        "$vtPath = $dir.Replace('\\', '\\\\')",
+        "$regContent = (Get-Content \"$bucketsdir\\extras\\scripts\\vt-uploader\\install-context.reg\").Replace('$dir', \"$vtPath\")",
+        "Set-Content \"$dir\\install-context.reg\" -Value $regContent -Encoding 'Ascii'"
+    ],
```
Comment (HUMORCE) [inline bucket/vt-uploader.json:25]:
```suggestion
    "post_install": [
        "$vtPath = $dir.Replace('\\', '\\\\')",
        "$regContent = (Get-Content \"$bucketsdir\\nuke\\scripts\\vt-uploader\\install-context.reg\").Replace('$dir', \"$vtPath\")",
        "if ($global) { $regContent = $regContent.Replace('HKEY_CURRENT_USER', 'HKEY_LOCAL_MACHINE') }",
        "Set-Content \"$dir\\install-context.reg\" -Value $regContent -Encoding 'Ascii'",
        "reg import \"$dir\\install-context.reg\""
    ],
```

File: bucket/vt-uploader.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    "homepage": "https://github.com/SamuelTulach/VirusTotalUploader",
+    "description": "A simple application written in C# that allows for file uploading to the VirusTotal website. (Requires a Virustotal API Key)",
+    "license": "GPL-3.0",
+    "notes": [
+        "You can add VirusTotalUploader as a context menu option by running,",
+        "reg import \"$dir\\install-context.reg\"",
+        "",
+        "For a guide on how to get your Virustotal API Key, visit,",
+        "https://github.com/SamuelTulach/VirusTotalUploader#how-to-get-api-key"
+    ],
```
Comment (HUMORCE) [inline bucket/vt-uploader.json:12]:
```suggestion
    "notes": [
        "For a guide on how to get your Virustotal API Key, visit,",
        "https://github.com/SamuelTulach/VirusTotalUploader#how-to-get-api-key"
    ],

```

File: bucket/vt-uploader.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            "VirusTotal Uploader"
+        ]
+    ],
+    "persist": "vtu_settings.json",
+    "pre_uninstall": [
+        "if (!(is_admin)) {error \"$app requires admin rights to $cmd\"; break}",
+        "if ($cmd -eq 'uninstall') {",
+        "   reg delete 'HKCR\\*\\shell\\Upload to VirusTotal' /f",
+        "}"
+    ],
```
Comment (HUMORCE) [inline bucket/vt-uploader.json:44]:
```suggestion
    "pre_uninstall": [
        "$registry = \"HKCU\\Software\\Classes\\*\\shell\\Upload to VirusTotal\"",
        "if ($cmd -eq 'uninstall') {",
        "    if ($global) { reg delete $registry.Replace('HKCU', 'HKLM') /f }",
        "    else { reg delete $registry /f }",
        "}"
    ],
```

**Review by HUMORCE — CHANGES_REQUESTED** [review]
context menu without elevated privileges, and consider of global installation.

I am not sure if this works with developer mode disabled. (Windows Settings > Privacy & security > For developer > Developer Mode )

**Comment by HUMORCE** [comment]
you might need to recreate a plain text format of `install-context.reg` replace current one, it's content cannot be tracked in the file changes.

File: bucket/vt-uploader.json (hunk 4 above)
Comment (Zliced13) [reply bucket/vt-uploader.json:12]: Why would I get rid of the notes that tell the user how to add `vt-uploader` as a context menu option.

Comment (HUMORCE) [reply bucket/vt-uploader.json:12]: because it does not require elevated privileges now

**Comment by Zliced13** [comment]
> context menu without elevated privileges, and consider of global installation.
> 
> `install-context.reg` needs to changed to:
> 
> ```
> Windows Registry Editor Version 5.00
> 
> [HKEY_CURRENT_USER\Software\Classes\*\shell\Upload to VirusTotal\command]
> @="\"$dir\\uploader.exe\" \"%1\""
> ```
> 
> I am not sure if this works with developer mode disabled. (Windows Settings > Privacy & security > For developer > Developer Mode )

Done, I changed it.

**Comment by HUMORCE** [comment]
screenshot_20221127180221

fix it please

**Comment by Zliced13** [comment]
> screenshot_20221127180221
> 
> 
> 
> fix it please
> 
> 
> 
> suggestion:
> 
> create `uninstall-context.reg` when you do this.
> 
> 
> 
> 
> 
> sorry for I did not list all at once. thanks 

@HUMORCE download the file and you can see for yourself that I did change it.

https://github.com/ScoopInstaller/Extras/blob/174fb74f2ebd2374e0b8f8ffcfce43b34ad754ea/scripts/vt-uploader/install-context.reg

**Comment by HUMORCE** [comment]
yes, but it is identified as a `binary` file, so we cannot viewing its changes.

when correct:
https://github.com/ScoopInstaller/Main/pull/4106/files#diff-8fee256e0fd73758f66ee53fa4ddaff6d04e97bfcd861204f61aaf53a9284312

**Comment by Zliced13** [comment]
Done, @HUMORCE, The reg file is fixed now.

**Review by HUMORCE — APPROVED** [review]

**Comment by HUMORCE** [comment]
thanks for your work, I will refine this manifest when i am free.

**Pull request merged**